src/main/java/com/ratifire/devrate/service/interview/InterviewCompletionService.java

```diff
@@ -60,10 +67,42 @@ public String completeInterviewProcess(Meeting meeting) {
     Long interviewerFeedbackDetailId = interviewFeedbackDetailId.get("interviewerFeedbackId");
     notificationService.addInterviewFeedbackDetail(interviewer, interviewerFeedbackDetailId);
 
+    try {
+      zoomApiService.deleteMeeting(Long.parseLong(meeting.getId()));
+    } catch (ZoomApiException e) {
+      logger.error("Zoom API exception occurred while trying to delete the meeting "
+          + "with meetingId: {}. {}", meeting.getId(), e.getMessage());
+    }
     interviewService.deleteInterview(interview.getId());
     interviewRequestService.deleteInterviewRequests(
         List.of(interview.getInterviewerRequest().getId(),
             interview.getCandidateRequest().getId()));
     return "Interview process completed successfully!";
   }
+
+  /**
+   * Validates if the webhook is triggered at the right time.
+   *
+   * @param payload The event data containing information about the meeting.
+   * @return boolean indicating if the webhook can proceed with interview completion.
+   * @throws ZoomWebhookException If the data is invalid.
+   */
+  public boolean validateMeetingEndTime(WebHookRequest payload) throws ZoomWebhookException {
+    if (payload == null || payload.getPayload() == null || payload
+        .getPayload().getMeeting() == null) {
+      throw new ZoomWebhookException("Invalid webhook payload");
+    }
+
+    WebHookRequest.Payload.Meeting meeting = payload.getPayload().getMeeting();
+    long meetingId = Long.parseLong(meeting.getId());
+
+    Interview interview = interviewService.getInterviewByMeetingId(meetingId);
+    ZonedDateTime scheduledStartTime = interview.getStartTime();
+    ZonedDateTime currentDateTime = DateTimeUtils.convertToUtcTimeZone(ZonedDateTime.now());
+
+    if (currentDateTime.isBefore(scheduledStartTime.plusMinutes(10))) {
```

Comment:
number `10` should be the constant that declared at the top of the class

Reply:
In addition, we could simplify this code sniped. Just using     return !currentDateTime.isBefore(scheduledStartTime.plusMinutes(NAME_VARIABLE));

Reply:
> In addition, we could simplify this code sniped. Just using return !currentDateTime.isBefore(scheduledStartTime.plusMinutes(NAME_VARIABLE));

Yes, it's a great idea, I also wanted to point that out, but it decreases readability